public/docs/_examples/lifecycle-hooks/dart/lib/after_view_component.dart

```diff
@@ -1,77 +1,117 @@
+// #docplaster
 // #docregion
 import 'package:angular2/core.dart';
 
-import 'child_component.dart';
 import 'logger_service.dart';
 
+//////////////////
+// #docregion child-view
 @Component(
-    selector: 'after-view-parent',
-    template: '''
-    <div class="parent">
-      <h2>AfterView</h2>
-
-      <div>
-        <input [(ngModel)]="hero">
-        <button (click)="showChild = !showChild">Toggle child view</button>
-
-        <my-child *ngIf="showChild" [hero]="hero"></my-child>
-      </div>
+  selector: 'my-child',
+  template: '<input [(ngModel)]="hero">')
+class ChildViewComponent {
+  String hero = 'Magneta';
+}
+// #enddocregion child-view
 
-      <h4>-- Lifecycle Hook Log --</h4>
-      <div *ngFor="let msg of hookLog">{{msg}}</div>
-    </div>
-    ''',
-    styles: const [
-      '.parent {background: burlywood; padding: 8px; margin:100px 8px;}'
-    ],
-    directives: const [ChildComponent],
-    providers: const [LoggerService])
-class AfterViewParentComponent
-    implements AfterContentInit, AfterViewChecked, AfterViewInit {
+//////////////////////
+@Component(
+  selector: 'after-view',
+// #docregion template
+  template: '''
+    <div>-- child view begins --</div>
+      <my-child></my-child>
+    <div>-- child view ends --</div>
+    <p *ngIf="comment != null" class="comment">{{comment}}</p>''',
+// #enddocregion template
+  directives: const [ChildViewComponent])
+// #docregion hooks
+class AfterViewComponent implements AfterViewChecked, AfterViewInit {
+  var _prevHero = '';
+  
+  // Query for a VIEW child of type `ChildViewComponent`
+  @ViewChild(ChildViewComponent) ChildViewComponent viewChild;
+  
+// #enddocregion hooks
   LoggerService _logger;
-  List<String> hookLog;
-  String hero = 'Magneta';
-  bool showChild = true;
+  AfterViewComponent(this._logger) {
+    _logIt('AfterView constructor');
+  }
+
+// #docregion hooks
+  ngAfterViewInit() {
+    // viewChild is set after the view has been initialized
+    _logIt('AfterViewInit');
+    _doSomething();
+  }
 
-  // Query for a CONTENT child of type `ChildComponent`
-  // No such CONTENT child exists!
-  // This component holds a view but no content of that type.
-  @ContentChild(ChildComponent)
-  ChildComponent contentChild;
+  ngAfterViewChecked() {
```

Comment:
It has an implicit return type of `dynamic`, thus matching the signature of the interface method it is overriding. I would suggest leaving it like this.